cmd/mattermost/commands/db.go

```diff
@@ -133,20 +154,103 @@ func resetCmdF(command *cobra.Command, args []string) error {
 
 func migrateCmdF(command *cobra.Command, args []string) error {
 	cfgDSN := getConfigDSN(command, config.GetEnvironment())
+	recoverFlag, _ := command.Flags().GetBool("auto-recover")
+	savePlan, _ := command.Flags().GetBool("save-plan")
```

Comment:
Actually `--auto-recover` does not require the plan to exist, it will try to recover on the fly whenever it gets an error. We might want to separate plan and auto recovery. But we can set save-plan to true if auto-recover is enabled as the user might be thinking of a possible recovery.

Reply:
Sounds good to me